### PR TITLE
Add handler for generating and dispatching change of supplier integration event

### PR DESCRIPTION
**Pull request by kft**
## Description
This PR adds a notification handler for generating `EnergySupplierChanged` integration event.
There might be additional metadata needed for this integration event, but this can be added later on.

### CODE REVIEW

File: source/business-workflow/source/Energinet.DataHub.MarketRoles.Domain/MeteringPoints/Transaction.cs [hunk 1]
```diff
@@ -26,6 +26,11 @@ public Transaction(string value)
 
         public string Value { get; set; }
 
+        public static Transaction Create(string transction)
```
Comment (thbaan) [inline source/business-workflow/source/Energinet.DataHub.MarketRoles.Domain/MeteringPoints/Transaction.cs:29]: Typo in 'transaction'

File: source/business-workflow/source/Energinet.DataHub.MarketRoles.Infrastructure/IntegrationEventDispatching/ChangeOfSupplier/PublishWhenEnergySupplierHasChanged.cs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+                supplierGlnNumber,
+                notification.StartOfSupplyDate);
+
+            var message = _outboxMessageFactory.CreateFrom(integrationEvent, OutboxMessageCategory.IntegrationEvent);
+            _outbox.Add(message);
+        }
+
+        private async Task<string> GetSupplierGlnNumberAsync(EnergySupplierId energySupplierId)
+        {
+            var sql = $"SELECT GlnNumber FROM [dbo].[EnergySuppliers] WHERE Id = @EnergySupplierId";
```
Comment (thbaan) [inline source/business-workflow/source/Energinet.DataHub.MarketRoles.Infrastructure/IntegrationEventDispatching/ChangeOfSupplier/PublishWhenEnergySupplierHasChanged.cs:55]: We should make an abstraction here to decouple logic for making DB requests from this Notification Handler.

**Review by thbaan — APPROVED** [review]

File: source/business-workflow/source/Energinet.DataHub.MarketRoles.Infrastructure/IntegrationEventDispatching/ChangeOfSupplier/PublishWhenEnergySupplierHasChanged.cs (hunk 2 above)
Comment (kft) [reply source/business-workflow/source/Energinet.DataHub.MarketRoles.Infrastructure/IntegrationEventDispatching/ChangeOfSupplier/PublishWhenEnergySupplierHasChanged.cs:55]: I don't see a need for that. It a simple read operation at infrastructure layer and the DB schema is owned by this BC. Could you elaborate a bit ?

File: source/business-workflow/source/Energinet.DataHub.MarketRoles.Domain/MeteringPoints/Transaction.cs (hunk 1 above)
Comment (kft) [reply source/business-workflow/source/Energinet.DataHub.MarketRoles.Domain/MeteringPoints/Transaction.cs]: Will fix it in a follow up PR.

**Pull request merged**